internal/charmstore/store.go

```diff
@@ -523,9 +523,12 @@ func (s *Store) AddCharm(c charm.Charm, p AddParams) (err error) {
 	// always be canonical, but check just in case anyway, as this is
 	// final gateway before a potentially invalid url might be stored
 	// in the database.
-	if p.URL.URL.Series == "bundle" || p.URL.URL.User == "" || p.URL.URL.Revision == -1 || p.URL.URL.Series == "" {
+	if p.URL.URL.Series == "bundle" || p.URL.URL.User == "" || p.URL.URL.Revision == -1 {
 		return errgo.Newf("charm added with invalid id %v", &p.URL.URL)
 	}
+	if p.URL.URL.Series == "" && len(c.Meta().Series) == 0 {
+		return errgo.Newf("charm added without series %v", &p.URL.URL)
```

Comment:
Also, where is it checked that once a Meta.Series exists in the latest revision, one can not upload old version of a charm without series in the metadata?